compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -132,7 +132,23 @@ fn get_llvm_object_symbols(
     if err.is_null() {
         return Ok(true);
     } else {
-        return Err(unsafe { *Box::from_raw(err as *mut io::Error) });
+        let error = unsafe { *Box::from_raw(err as *mut String) };
+        if buf.starts_with(&[0xDE, 0xCE, 0x17, 0x0B]) || buf.starts_with(b"BC\xC0\xDE") {
+            // For LLVM bitcode failure to read the symbols is not fatal. The bitcode may have been
```

Comment:
```suggestion
            // For LLVM bitcode, failure to read the symbols is not fatal. The bitcode may have been
```

Reply:
Added a reference to LLVM's detection code for bitcode files.